In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
pd.set_option('display.max_rows', 1000)

This Week - (15 JUL 2023 - 20 JUL 2023);
Last Week - (8 JUL 2023 - 13 JUL 2023)

In [17]:
page_data_last_week = pd.read_excel('page_data_last_week.xlsx')
page_data_last_week

Page  Pageviews  Unique Pageviews  Avg. Time on Page  \
0  /products/test-rings        342               183         153.165803   

   Entrances  Bounce Rate    % Exit  Page Value  
0        110     0.781818  0.435673     0.15847

In [18]:
campaign_data_this_week = pd.read_excel('ads_data_this_week.xlsx')
campaign_data_this_week

Google Ads: Ad Group  Clicks   Cost       CPC  Users  Sessions  Bounce Rate  \
0           Ad group 1     209  85.02  0.406794    195       208     0.884615   

   Pages / Session  Ecommerce Conversion Rate  Transactions  Revenue  
0         1.173077                          0             0        0

In [19]:
page_data_this_week = pd.read_excel('page_data_this_week.xlsx')
page_data_this_week

Page  Pageviews  Unique Pageviews  Avg. Time on Page  \
0  /products/test-rings       1129               301          67.409144   

   Entrances  Bounce Rate    % Exit  Page Value  
0        250        0.812  0.244464           0

In [20]:
keywords_this_week = pd.read_excel('keywords_data_this_week.xlsx')
keywords_this_week

Keyword  Clicks   Cost       CPC  Users  \
0           how to measure ring size      36  14.98  0.416111     33   
1               Printable ring sizer      33  13.01  0.394242     30   
2                    ring size chart      18   7.28  0.404444     16   
3                       ring measure      14   5.50  0.392857     17   
4                         ring sizer      14   5.82  0.415714     14   
5                        ring size 7      13   5.03  0.386923     11   
6   how to measure ring size at home      12   4.40  0.366667     12   
7                     mens ring size      11   4.61  0.419091     10   
8                  average ring size       6   2.41  0.401667      5   
9                    free ring sizer       6   2.21  0.368333      6   
10                               NaN     210  85.42  0.406762    197   

    Sessions  Bounce Rate  Pages / Session  Ecommerce Conversion Rate  \
0         34     0.970588         1.029412                          0   
1         34     0.911765         1.147059                          0   
2         17     0.882353         1.176471                          0   
3         18     0.888889         1.111111                          0   
4         14     0.785714         1.285714                          0   
5         12     1.000000         1.000000                          0   
6         14     0.928571         1.071429                          0   
7         11     0.909091         1.181818                          0   
8          5     0.600000         1.400000                          0   
9          6     0.666667         1.333333                          0   
10       209     0.885167         1.172249                          0   

    Transactions  Revenue  
0              0        0  
1              0        0  
2              0        0  
3              0        0  
4              0        0  
5              0        0  
6              0        0  
7              0        0  
8              0        0  
9              0        0  
10             0        0

JUST COMPARING PAGE DATA:
1) Views jumped from 342 to 1129 = 230% increase
2) Bounce Rate increased from 78.18 to 81.2 - 3.02% increase
This means even when more people clicked on the site, more people did not like what they saw, meaning they found stuff irrelevant.
3) Average time on page decreased from 153.16 to 67.40  - 55.99 % decrease
4) %exit decreased from 0.43 to 0.24 - 44.19%
Bounce rate is people opened the page and left the same page.
Exit rate is people started from somewhere else but ended from here.

In the current week,
1. we had 820 organic views.
2. The bounce rate is high 88.46%. This means 185 approx people left the site without any engagement thorugh ads. For the whole page, 917 people left immediately. So, organically, 917-185 = 732 people left fast.
Interestingly, prior to Saturday and after Saturday bounce rate for ads was almost same

Looks like people are not still satisfied with - 'how to measure ring size', 'how to measure ring size at home', 'mens ring size'. This means maybe people are not browsing anymore. 

Interestingly, 'average ring size' and 'free ring sizer' showed low bounce rates. Maybe  due to less volume or maybe article answers this better
